src/js/tech/html5.js

```diff
@@ -203,6 +207,31 @@ class Html5 extends Tech {
     });
   }
 
+  /**
+   * Handle IOS Headphone disconnection during playback
+   *
+   * @private
+  */
+  handleIOSHeadphonesDisconnection_() {
+
+    const TIME_FUDGE_FACTOR = 1 / 30;
+
+    const SAFE_TIME_DELTA = TIME_FUDGE_FACTOR * 3;
+
+    const disconnectionIOSListener = () => {
+      if ((!this.el_.paused && window.navigator.onLine) &&
+          (this.el_.buffered.end(this.el_.buffered.length - 1) + SAFE_TIME_DELTA >= this.el_.currentTime)) {
```

Comment:
Also, it's possible that there are multiple buffered regions, and we're current in the first or a middle buffered region instead of the last one.

Reply:
All good points!

Reply:
@gkatsev Can you explain a little more, since we try to get the end time buffered. (length - 1) and if we have multiple regions it does not matter that means that we still have buffered time and we are not experiencing a `Stalled` or `suspend` due another problem

Reply:
@marcodeltorob basically, we could be at the end of a buffered region which isn't the last buffered region available, so, currentTime will end up being less than the buffered region.
For example, we're at 10 seconds in (marked by `|`, and our buffer looks something like:
```
[0-10|----20-30----60-70----]
```

Reading the last buffered region, would give us 70 but our currentTime is 10. However, we still can't play because we've run out of buffer.
But, if the currentTime is say 5, and our buffered region has extra buffer, then we should be fine.
```
[0-5|5-10----20-30----60-70----]
```